### PR TITLE
RSDK-6373 Add wrappers for generic service

### PR DESCRIPTION
**Pull request by benjirewis**
RSDK-6373

Adds client and server wrappers for the new generic service protos. Adds test utilities and tests for generic services. Modifies generic component language to make distinction between the two more clear.

### CODE REVIEW

File: go.mod
Comment (benjirewis) [inline go.mod:401]: To use https://github.com/viamrobotics/api/pull/430. Hold off on reviewing this until those proto changes are merged.

File: robot/client/client_test.go
Comment (benjirewis) [inline robot/client/client_test.go:1069]: See my TODO comment above. I broke this test and gRPC reflection for `robot.proto` by trying to import `app.proto` in `robot.proto` 😮‍💨 . It should be fixed with https://github.com/viamrobotics/api/pull/429, but until then, we should probably skip this test, as it will fail in any RDK PR attempting to upgrade the version of the `go.viam.com/api` dependency.

File: services/generic/client.go
Comment (benjirewis) [inline services/generic/client.go:1]: The generic service code and tests are pretty much copy-pasted from the generic component code with `:%s/component*/service*/g`. Let me know if you think I shouldn't be copy-pasting anywhere...

File: services/generic/fake/generic.go
Comment (maximpertsov) [inline services/generic/fake/generic.go:13]:
```suggestion
	resource.RegisterService(
```

Comment (benjirewis) [reply services/generic/fake/generic.go]: 😮‍💨 good catch! Ran an RDK server with the following config to ensure things built correctly:
```json
{
  "services": [
    {
      "name": "g",
      "type": "generic",
      "model": "fake"
    }
  ]
}
```

File: robot/client/client_test.go
Comment (benjirewis) [reply robot/client/client_test.go:1069]: Yeah that would be cool 🤔 . We tried to do this in the Mongo Go driver, too, but found it was better to assert that things failed in a particular way (i.e. a test that passes when certain things fail) rather than ensure that `go test` fails a certain `t.Run`.

**Review by maximpertsov — APPROVED** [review]

**Review by maximpertsov — APPROVED** [review]
LGTM minus the `go.mod/go.sum` conflicts

**Comment by benjirewis** [comment]
We can upgrade to the [newest](https://github.com/viamrobotics/api/commit/9f36d411077598f88db0cb2d9b0faf01327848e0) `api` version and _not_ skip `TestClientResources`/break Golang SDK reflection. But, we'll need to wait for a couple PRs that [mark usage of deprecated sensors method with `nolint`](https://github.com/viamrobotics/rdk/pull/3461), and [remove `GetLatestMapInfo` from SLAM](https://github.com/viamrobotics/rdk/pull/3388) to be merged.

File: services/generic/fake/generic.go
Comment (cheukt) [inline services/generic/fake/generic.go:30]: does this fake generic even echo inputs? I don't see a DoCommand anywhere

File: services/generic/register/register.go
Comment (cheukt) [inline services/generic/register/register.go:1]: also need to register in services/register/all.go

File: services/generic/fake/generic.go
Comment (benjirewis) [reply services/generic/fake/generic.go:30]: It does not; good catch. I copied this comment from the fake generic component, which also does not actually echo inputs back to the caller. Modified the comment for both. If you think it's important functionality, I can also make them actually echo inputs.

File: services/generic/register/register.go
Comment (benjirewis) [reply services/generic/register/register.go:1]: Ooh good catch; didn't know about that top level registration.

File: services/generic/fake/generic.go
Comment (cheukt) [reply services/generic/fake/generic.go]: ya, let's make them actually echo inputs - hopefully should be pretty quick!

Comment (benjirewis) [reply services/generic/fake/generic.go]: Can do.

Comment (benjirewis) [reply services/generic/fake/generic.go]: Added `DoCommand` to both fake generic services and components and added small tests for them.

**Review by cheukt — APPROVED** [review]
LGTM!

**Pull request merged**
